php/class-post-type.php

```diff
@@ -675,4 +711,68 @@ public function hide_disabled_publishing_actions( $post ) {
 		</style>
 		<?php
 	}
+
+	/**
+	 * Get conflicts settings
+	 *
+	 * @param \WP_Post $post post to compare conflict values.
+	 * @param array    $settings setting to search for optional.
+	 *
+	 * @return array
+	 */
+	public function get_conflicts_setting( $post, $settings = array() ) {
+		global $wpdb;
+		if ( $post ) {
+			$post = get_post( $post );
+		}
+		if ( empty( $settings ) ) {
+			$content = $this->get_post_content( $post );
+			$return = array();
+			if ( empty( $content ) || ! is_array( $content ) ) {
+				return $return;
+			}
+			$settings = array_keys( $content );
+			if ( empty( $settings ) ) {
+				return $return;
+			}
+		}
+		$return = array();
+		$query = $wpdb->prepare( "SELECT ID, post_title, post_status, post_content FROM $wpdb->posts WHERE post_type = %s AND post_status IN ( 'draft', 'pending', 'future' ) ", static::SLUG );
+		// Todo: finalize post_status to check in.
+		if ( $post instanceof \WP_Post ) {
+			$query .= $wpdb->prepare( 'AND ID != %d ', $post->ID );
+		}
+		$query .= 'AND ( ';
+		$or = array();
+		foreach ( $settings as $setting_id ) {
+			$or[] = $wpdb->prepare( 'post_content LIKE %s', '%' . $wpdb->esc_like( wp_json_encode( $setting_id ) ) . '%' );
+		}
+		$query .= implode( ' OR ', $or );
+		$query .= ' )';
+
+		$results = $wpdb->get_results( $query, ARRAY_A ); // WPCS: unprepared SQL ok.
+
+		if ( ! empty( $results ) ) {
+			foreach ( $results as $item ) {
+				$data = json_decode( $item['post_content'], true );
+				$snapshot_content_keys = array_keys( $data );
+				$conflicts_keys = array_intersect( $snapshot_content_keys, $settings );
+				if ( empty( $conflicts_keys ) ) {
+					continue;
+				}
+				foreach ( $conflicts_keys as $conflicts_key ) {
+					if ( ! isset( $return[ $conflicts_key ] ) ) {
+						$return[ $conflicts_key ] = array();
+					}
+					$return[ $conflicts_key ][] = array(
+						'ID' => $item['ID'],
+						'value' => $data[ $conflicts_key ]['value'],
+						'name' => $item['post_title'],
+						'editLink' => get_edit_post_link( $item['ID'], 'raw' ),
```

Comment:
PHP keys should be `snake_case` I think. So `id` instead of `ID` and `edit_link` instead of `editLink`.